admin_app/src/app/content/page.tsx

```diff
@@ -316,34 +320,45 @@ const CardsGrid = ({
   };
 
   React.useEffect(() => {
-    apiCalls
-      .getContentList({ token: token!, skip: 0, limit: MAX_CARDS_TO_FETCH })
-      .then((data) => {
-        const filteredData = data.filter((card: Content) => {
-          const matchesSearchTerm =
-            card.content_title
-              .toLowerCase()
-              .includes(searchTerm.toLowerCase()) ||
-            card.content_text.toLowerCase().includes(searchTerm.toLowerCase());
+    if (token) {
+      apiCalls
+        .getContentList({ token: token!, skip: 0, limit: MAX_CARDS_TO_FETCH })
+        .then((data) => {
+          const filteredData = data.filter((card: Content) => {
+            const matchesSearchTerm =
+              card.content_title
+                .toLowerCase()
+                .includes(searchTerm.toLowerCase()) ||
+              card.content_text
+                .toLowerCase()
+                .includes(searchTerm.toLowerCase());
 
-          const matchesAllTags = filterTags.some((fTag) =>
-            card.content_tags.includes(fTag.tag_id)
-          );
+            const matchesAllTags = filterTags.some((fTag) =>
+              card.content_tags.includes(fTag.tag_id),
+            );
 
-          return (
-            matchesSearchTerm && (filterTags.length === 0 || matchesAllTags)
-          );
-        });
+            return (
+              matchesSearchTerm && (filterTags.length === 0 || matchesAllTags)
+            );
+          });
 
-        setCards(filteredData);
-        setMaxPages(Math.ceil(filteredData.length / MAX_CARDS_PER_PAGE));
-        setIsLoading(false);
-      })
-      .catch((error) => {
-        console.error("Failed to fetch content:", error);
-        setSnackMessage({ message: `Failed to fetch content`, color: "error" });
-        setIsLoading(false);
-      });
+          setCards(filteredData);
+          setMaxPages(Math.ceil(filteredData.length / MAX_CARDS_PER_PAGE));
+          setIsLoading(false);
+        })
+        .catch((error) => {
+          console.error("Failed to fetch content:", error);
+          setSnackMessage({
+            message: `Failed to fetch content`,
+            color: "error",
+          });
+          setIsLoading(false);
+        });
+    } else {
+      setCards([]);
+      setMaxPages(1);
+      setIsLoading(false);
+    }
```

Comment:
I figured it is safer to set the data to null, even though shortly after we get redirected to login page. Wdyt?

Reply:
Im ok with it. Thanks!